src/node_jsvmapi.cc

```diff
@@ -465,22 +465,109 @@ namespace v8impl {
 
 }  // end of namespace v8impl
 
-// We define a macro here because, with full error handling, we may want to
-// perform additional actions at the top of our public APIs, such as bail,
-// returning an error code at the very top of each public API in the case that
-// there's already an exception pending in v8impl::TryCatch::lastException
-#define NAPI_PREAMBLE(e) \
-  v8impl::TryCatch tryCatch(v8impl::V8IsolateFromJsEnv((e)))
+#define RETURN_STATUS_IF_FAIL(condition, status)                        \
```

Comment:
I'd name this `RETURN_STATUS_IF_FALSE` or `..._IF_NOT`, since really it checks if the condition is false, not if it "failed".